src/hooks/useMenuKeyboardNavigation.ts

```diff
@@ -0,0 +1,88 @@
+import React from 'react'
+import {iterateFocusableElements} from '@primer/behaviors/utils'
+import {useMenuInitialFocus} from './useMenuInitialFocus'
+import {useTypeaheadFocus} from './useTypeaheadFocus'
+import {MenuContextProps} from '../ActionMenu'
+
+/**
+ * Keyboard navigation is a mix of 4 hooks
+ * 1. useMenuInitialFocus
+ * 2. useTypeaheadFocus
+ * 3. useCloseMenuOnTab
+ * 4. useMoveFocusToMenuItem
+ */
+export const useMenuKeyboardNavigation = (
+  open: boolean,
+  onOpen: MenuContextProps['onOpen'],
+  onClose: MenuContextProps['onClose'],
+  containerRef: React.RefObject<HTMLElement>,
+  anchorRef: React.RefObject<HTMLElement>
+) => {
+  const {openWithFocus} = useMenuInitialFocus(open, onOpen, containerRef, anchorRef)
+  useTypeaheadFocus(open, containerRef, anchorRef)
+  useCloseMenuOnTab(open, onClose, containerRef, anchorRef)
+  useMoveFocusToMenuItem(open, containerRef, anchorRef)
+
+  return {openWithFocus}
+}
+
+/**
+ * When Tab or Shift+Tab is pressed, the menu should close
+ * and the focus should naturally move to the next item
+ */
+const useCloseMenuOnTab = (
```

Comment:
before, ignore `Tab` when the menu is open:

https://user-images.githubusercontent.com/1863771/159462565-f0a7527b-4cf1-4a0c-a21b-5c1ada74e14d.mov

after, pressing `Tab` should close the Menu and let the focus go to the next item (or previous item for `Shift+Tab`)

https://user-images.githubusercontent.com/1863771/159462569-aaf9ba21-f061-42cc-adb4-43d08995dbf7.mov